dspace-api/src/main/java/org/dspace/app/sherpa/SHERPAService.java

```diff
@@ -91,12 +108,22 @@ public SHERPAResponse searchByJournalISSN(String query) {
                 HttpEntity responseBody = response.getEntity();
 
                 if (null != responseBody) {
-                    sherpaResponse = new SHERPAResponse(responseBody.getContent());
+                    InputStream content = null;
+                    try {
+                        content = responseBody.getContent();
+                        sherpaResponse = new SHERPAResponse(content, SHERPAResponse.SHERPAFormat.JSON);
+                    } catch (IOException e) {
+                        log.error("Encountered exception while contacting SHERPA/RoMEO: " + e.getMessage(), e);
+                    } finally {
+                        if (content != null) {
+                            content.close();
+                        }
+                    }
                 } else {
                     sherpaResponse = new SHERPAResponse("SHERPA/RoMEO returned no response");
                 }
             } catch (Exception e) {
-                log.warn("Encountered exception while contacting SHERPA/RoMEO: " + e.getMessage(), e);
+                log.error("Encountered exception while contacting SHERPA/RoMEO: " + e.getMessage(), e);
```

Comment:
We have to handle at least 3 exception types here (Interrupted, for sleep, URISyntax, and IO) - and so I guess rethrow each type? Or is there a new exception type you think I should throw?
Unlike the provider classes, this one is OK to change in terms of exception usage since it isn't currently used by anything (presumably the REST API can use it in future), and doesn't implement any superclasses.
If you want me to rethrow each exception type in its own catch, that's fine. If you want me to use Exception, that's fine too. There must be one that works best though... HttpResponseException?